### PR TITLE
Stac 0 re improve molecule

### PR DESCRIPTION
**Pull request by MelcomVE**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: .gitlab-ci-agent.yml [hunk 1]
```diff
@@ -630,22 +624,22 @@ pre_release_trace_agent_image:
...
   rules:
     - *only_agent_3
 
-## acceptance stage
-
-.molecule_base: &molecule_base
-  stage: acceptance
+# Core Molecule Structure for all the molecule steps
+.molecule: &molecule
+  interruptible: true
```
Comment (craffit) [inline .gitlab-ci-agent.yml:640]: interruptible is dangerous.. it can cause cleanup to not properly work

File: .gitlab-ci-agent.yml [hunk 2]
```diff
@@ -655,92 +649,175 @@ pre_release_cluster_agent_image:
...
+    - molecule_compose_test
+    - molecule_compose_create
+  needs:
+    - molecule_compose_test
+    - molecule_compose_create
+
+# Job Seq: Molecule
+# Type: Integrations
+
+.molecule_integrations_rules: &molecule_integrations_rules
```
Comment (craffit) [inline .gitlab-ci-agent.yml:751]: it looks a bit weird to me that we have 4 conditions defined here, that fire based on where they are included.. why not split out the rules over the jobs?

I cannot decypher the commit message matching logic, what runs under what conditions with the commit message? Best add a comment to explain that (maybe even the README.md if it is a feature)

File: .gitlab-ci-agent.yml [hunk 3]
```diff
@@ -751,106 +828,264 @@ molecule_tests_kubernetes:
...
-  <<: *molecule_base
-  stage: acceptance_cleanup
-  script:
-    - cd $CI_PROJECT_DIR/test && ./molecule3.sh destroy -s swarm
-  needs:
-    - molecule_tests_swarm
-  allow_failure: true
+
+# Job Seq: Molecule
+# Type: localinstall
```
Comment (craffit) [inline .gitlab-ci-agent.yml:846]: what is localinstall?

Comment (craffit) [reply .gitlab-ci-agent.yml:846]: ah i see, its a suite

File: .gitlab-ci-agent.yml (hunk 2 above)
Comment (craffit) [reply .gitlab-ci-agent.yml:751]: I found the comment on .gitlab-ci.yml, that looks great! Still the logic looks convoluted, why 4 conditionals all together?

File: test/molecule3.sh [hunk 4]
```diff
@@ -42,7 +49,12 @@ fi
 
 cd molecule-role
 
+# Allows the yaml to be tested before spinning up and instance
+yamllint -c .yamllint .
```
Comment (craffit) [inline test/molecule3.sh:53]: will people always have this installed locally? how will we get it?

File: .gitlab-ci-agent.yml (hunk 2 above)
Comment (MelcomVE) [reply .gitlab-ci-agent.yml:751]: True true, I might be able to make these rules more compact. Looking at this now

**Review by craffit — APPROVED** [review]

**Pull request merged**